### PR TITLE
[4.0] Move component specific fields out of com_menus

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes

Moves component-specific field class out of `com_menus` and into respective components.
Also corrects class/filename casing.

### Testing Instructions

Create administrator menu without preset.

Go to create a new menu item.

Select `Categories -> Create New Category` and `Categories -> List All Categories` menu item types. In their forms check that `Choose a Component` dropdown appears.

Select `Fields -> List All Field Groups`, `Fields -> List All Fields` and `Workflows -> List All Workflows` menu item types. In their forms check that `Context` dropdown appears.

### Expected result

Works like before.

### Documentation Changes Required

IDK.

### CODE REVIEW

File: administrator/components/com_categories/tmpl/categories/default.xml [hunk 1]
```diff
@@ -12,6 +12,7 @@
 				type="ComponentsCategory"
 				label="COM_CATEGORIES_CHOOSE_COMPONENT_LABEL"
 				required="true"
+				addfieldprefix="Joomla\Component\Categories\Administrator\Field"
 			>
```
Comment (Quy) [inline administrator/components/com_categories/tmpl/categories/default.xml]:
```suggestion
				>
```

**Comment by ceford** [comment]
I could not test this patch because I could not find the items mentioned in the test instructions. I created an empty menu and then a New menu item. But there is no Categories -> Create New Category link and I could not find anything similar. Please provide more specific instructions. I am using the Joomla 4 Patch Tester.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28931">issues.joomla.org/tracker/joomla-cms/28931</a>.</sub>

**Comment by SharkyKZ** [comment]
@ceford You have to create a menu for administrator, not for site. To do that, click Menus -> Manage. And then select Administrator in the dropdown.

**Comment by ceford** [comment]
Tested with the following snag: the Fields -> List All Field Groups and Fields -> List All Fields items in the Menu Items list do not actually appear in the Test menu I created. Is that a bug here or a menu bug? <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28931">issues.joomla.org/tracker/joomla-cms/28931</a>.</sub>

**Comment by SharkyKZ** [comment]
I'm not sure. But for the test you don't need to create menu items. Only open the menu item form, select correct menu item type and check that the mentioned fields appear, e.g. like this context field:

Screenshot_2020-05-05 Menus New Item - Joomla - Administration

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on d920fed2ad45d39fb13c95f0a5f0e4e8d5a7e7cd<br /><br />It does do what the test criteria say it should do. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28931">issues.joomla.org/tracker/joomla-cms/28931</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on d920fed2ad45d39fb13c95f0a5f0e4e8d5a7e7cd<br /><br />Tested successfully in Nightly Build of 15 May<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28931">issues.joomla.org/tracker/joomla-cms/28931</a>.</sub>

**Comment by SharkyKZ** [comment]
RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28931">issues.joomla.org/tracker/joomla-cms/28931</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**